src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableExtensions.cs

```diff
@@ -17,6 +17,9 @@ internal static partial class ImmutableExtensions
     {
         internal static bool IsValueType<T>()
         {
+#if NETCOREAPP
+            return typeof(T).IsValueType;
+#else
```

Comment:
Can the !NETCOREAPP path be replaced with:
```
return (default(T) != null) ? true : typeof(T).IsValueType;
```

Reply:
Tried this, but it turns out the _System.Collections.Immutable_ package targets netstandard1.0. The `Type.IsValueType` API wasn't introduced until netstandard2.0.

I can try using `typeof(T).GetTypeInfo().IsValueType`, but this would also require me to add a reference to the _System.Reflection_ package. And I'm afraid it's now becoming a more invasive change.

Should I still pursue this suggestion?

Reply:
I do not think it is worth. I would leave it as it is.

Reply:
Ah well. Thanks for the suggestion anyway! :)